crates/revm/src/bsc/handler_register.rs

```diff
@@ -0,0 +1,65 @@
+//! Handler related to BNB-Smart-chain
+
+use crate::{
+    handler::register::EvmHandler,
+    interpreter::Gas,
+    primitives::{address, db::Database, spec_to_generic, Address, EVMError, Spec, SpecId, U256},
+    Context,
+};
+use std::sync::Arc;
+
+pub const SYSTEM_ADDRESS: Address = address!("fffffffffffffffffffffffffffffffffffffffe");
+
+pub fn bsc_handle_register<DB: Database, EXT>(handler: &mut EvmHandler<'_, EXT, DB>) {
+    spec_to_generic!(handler.cfg.spec_id, {
+        handler.post_execution.reward_beneficiary =
+            Arc::new(collect_system_reward::<SPEC, EXT, DB>);
+    });
+}
+
+/// Collect gas fee to system account.
+#[inline]
+pub fn collect_system_reward<SPEC: Spec, EXT, DB: Database>(
+    context: &mut Context<EXT, DB>,
+    gas: &Gas,
+) -> Result<(), EVMError<DB::Error>> {
+    let effective_gas_price = context.evm.env.effective_gas_price();
```

Comment:
Please pay attention to  `effective_gas_price` as the base fee of BSC is always zero.

Reply:
```
    pub fn effective_gas_price(&self) -> U256 {
        if let Some(priority_fee) = self.tx.gas_priority_fee {
            min(self.tx.gas_price, self.block.basefee + priority_fee)
        } else {
            self.tx.gas_price
        }
    }
```